hi-ml/src/health_ml/model_trainer.py

```diff
@@ -140,9 +140,10 @@ def create_lightning_trainer(container: LightningContainer,
         else:
             callbacks.append(more_callbacks)  # type: ignore
     callbacks.extend(container.get_callbacks())
-    profiler_args = {}
-    if "profiler" in additional_args:
-        profiler_args = additional_args.pop("profiler")
+    # Set profiler: if --pl_profiler=profiler is specified, it overrides any custom profiler defined in the container
+    custom_profiler = additional_args.pop("profiler", PassThroughProfiler())
```

Comment:
nitpicking - using PassThroughProfiler instead of None requires a lot of complicated checks, is that really necessary?

Reply:
Yes, this is what PL trainer uses by default. I changed get_pl_profiller also to return PassThroughProfiler as default.

https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.profiler.PassThroughProfiler.html

Reply:
We can also use None for custom_profiler but it complicates the number of checks to make to pick either the pl_profiler or the custom one. 

Or do you prefer setting defaults to None for both custom and pl_profiler and adding a check line 185, `profiler=profiler if profiler else PassTroughProfiler()` for the trainer

Reply:
Oh no, my bad if we give it None pl will set it to PassTroughProfiler. it's like when container.pl_profiler was None by default, will fix that.